### ISSUE TITLE
Q: Cause of 'Resolved include path exceeds file system root' warnings?

### ISSUE THREAD

**Issue by KevinMitchell** [mulesoft/api-designer#469]
I'm getting these warning in a few places in my specification.  But I don't understand what causes such warnings; what's it trying to tell me?  Is there a simple example that illustrates the problem and solution? 

Thanks

**Comment by sichvoge** [comment]
Hi Kevin. Can you attach an image of that warning pls?

**Comment by leealt** [comment]
Has anyone answered this? I am experiencing this also.

**Comment by sichvoge** [comment]
No one came back to my question. If you can attach a small example and an image that shows the warning, it would be very helpful.

**Comment by KevinMitchell** [comment]
Sorry for not replying; I moved across to using the API Workbench in Atom which avoided the problem.   Hopefully leealt can provide an example/image.

**Comment by dylan-asos** [comment]
I'm also getting these warnings.  I have a root API RAML document, with a number of directories, e..g

api.raml
---->traits
---->types
---->libraries
---->examples

In my library.raml file in the libraries directory, I'm referencing the individual files like so,

```yaml
#%RAML 1.0 Library 
usage: Defines the types for the api
types:  
  SomeType: !include ../types/SomeType.raml
traits:
  errorResponse: !include ../traits/errorResponse.raml
```
When I use the library from api.raml, and in particular the usage of the traits I've defined, I see the warnings next to my endpoints

![alt text](http://i65.tinypic.com/350jxur.jpg "Logo Title Text 1")

Everything seems to be working, I just see lots of warnings.

**Comment by dylan-asos** [comment]
I've restructured my directories like so, 

api.raml
---->libraries
-------->traits
-------->types
-------->examples

and my library definition

```yaml
#%RAML 1.0 Library 
usage: Defines the types for the api
types:  
  SomeType: !include types/SomeType.raml
traits:
  errorResponse: !include traits/errorResponse.raml
```

This has fixed the warnings

**Comment by leealt** [comment]
This example is right from theAdvanced  API Design training class

Create the matching directories in API Designer and import/move the files into the appropriate folders

Lee

**Comment by deividaspetraitis** [comment]
Bump. Same here:

![screenshot_19](https://user-images.githubusercontent.com/5105752/29166149-ee2ded76-7dcd-11e7-8c01-4601f3e39691.png)

**Comment by sichvoge** [comment]
Can you send us a small reproducable project, please?

**Comment by deividaspetraitis** [comment]
Sure, you can find whole API definition here: https://anypoint.mulesoft.com/apiplatform/noname-10/#/portals/organizations/eb83f888-dced-479c-8339-9a96f8f854e2/apis/22427119/versions/784156/pages/269641

Also I have noticed some weird behaviour for same API definition ( RAML is valid since it's the same project ) in api-designer. Where I should report about it:

![screenshot_20](https://user-images.githubusercontent.com/5105752/29202008-87b99be0-7e6c-11e7-8aa8-e8a40a5640f0.png)

**Comment by sichvoge** [comment]
Have you raised a support ticket with MuleSoft?

**Comment by deividaspetraitis** [comment]
Unfortunately I don't have account for support and can't create a ticket..

**Comment by eduardo-rubio** [comment]
https://www.mulesoft.org/jira/browse/DESIGNER-1734

**Comment by carowright** [comment]
The fix is available from version 0.4.12
